pkg/apis/keycloak/v1alpha1/keycloakclient_types.go

```diff
@@ -149,6 +155,147 @@ type KeycloakProtocolMapper struct {
+	// ID.
+	// +optional
+	ID string `json:"id,omitempty"`
+	// Name.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// Policies.
+	// +optional
+	Policies []KeycloakPolicy `json:"policies,omitempty"`
+	// The policy enforcement mode dictates how policies are enforced when evaluating authorization requests.
+	// 'Enforcing' means requests are denied by default even when there is no policy associated with a given resource.
+	// 'Permissive' means requests are allowed even when there is no policy associated with a given resource.
+	// 'Disabled' completely disables the evaluation of policies and allows access to any resource.
+	// +optional
+	PolicyEnforcementMode string `json:"policyEnforcementMode,omitempty"`
+	// Resources.
+	// +optional
+	Resources []KeycloakResource `json:"resources,omitempty"`
+	// Authorization Scopes.
+	// +optional
+	Scopes []KeycloakScope `json:"scopes,omitempty"`
+}
+
+type KeycloakPolicy struct {
+	// Config.
+	// +optional
+	Config map[string]string `json:"config,omitempty"`
+	// The decision strategy dictates how the policies associated with a given permission are evaluated and how
+	// a final decision is obtained. 'Affirmative' means that at least one policy must evaluate to a positive
+	// decision in order for the final decision to be also positive. 'Unanimous' means that all policies must
+	// evaluate to a positive decision in order for the final decision to be also positive. 'Consensus' means
+	// that the number of positive decisions must be greater than the number of negative decisions. If the number
+	// of positive and negative is the same, the final decision will be negative.
+	// +optional
+	DecisionStrategy string `json:"decisionStrategy,omitempty"`
+	// A description for this policy.
+	// +optional
+	Description string `json:"description,omitempty"`
+	// ID.
+	// +optional
+	ID string `json:"id,omitempty"`
+	// The logic dictates how the policy decision should be made. If 'Positive', the resulting effect
+	// (permit or deny) obtained during the evaluation of this policy will be used to perform a decision.
+	// If 'Negative', the resulting effect will be negated, in other words, a permit becomes a deny and vice-versa.
+	// +optional
+	Logic string `json:"logic,omitempty"`
+	// The name of this policy.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// Owner.
+	// +optional
+	Owner string `json:"owner,omitempty"`
+	// Policies.
+	// +optional
+	Policies []string `json:"policies,omitempty"`
+	// Resources.
+	// +optional
+	Resources []string `json:"resources,omitempty"`
+	// Resources Data.
+	// +optional
+	ResourcesData []KeycloakResource `json:"resourcesData,omitempty"`
+	// Scopes.
+	// +optional
+	Scopes []string `json:"scopes,omitempty"`
+	// Scopes Data.
+	// +optional
+	ScopesData []KeycloakScope `json:"scopesData,omitempty"`
+	// Type.
+	// +optional
+	Type string `json:"type,omitempty"`
+}
+
+type KeycloakResource struct {
+	// ID.
+	// +optional
+	ID string `json:"_id,omitempty"`
```

Comment:
I went with `_id` which is used within [ResourceRepresentation](https://github.com/keycloak/keycloak/blob/master/core/src/main/java/org/keycloak/representations/idm/authorization/ResourceRepresentation.java#L43) (which looks like a copypasta from some other `foo_id` field?)

Reply:
Yeah, when trying to use "id" here instead the REST API returns something along the lines of "this field is not marked as ignorable" because the JSON to Java Resource Object conversion fails because it wants "_id".

Reply:
Ok, thanks for checking!

Reply:
We use `_id` due to the alignment with UMA specs. It is OK.